src/details/ArborX_DetailsCrsGraphWrapperImpl.hpp

```diff
@@ -63,6 +64,15 @@ struct InsertGenerator
   using Access = AccessTraits<Predicates, PredicatesTag>;
   using PredicateType = typename AccessTraitsHelper<Access>::type;
 
+  // Legacy callback wrapper
+  template <typename Value, bool B = Legacy,
+            typename Enable = std::enable_if_t<!B>>
+  KOKKOS_FUNCTION auto operator()(PredicateType const &predicate,
+                                  Value const &value) const
+  {
+    return (*this)(predicate, (int)value.index);
+  }
+
   KOKKOS_FUNCTION auto operator()(PredicateType const &predicate,
                                   int primitive_index) const
   {
```

Comment:
This is still iffy. I think this will only work with `BasicBoundingVolumeHierarchy` that has `Value` hardcoded to `PairIndexVolume`. I hope to be able to switch the signature of the crs graph in the future, this should be the last hard task.